### ISSUE TITLE
Suggest split Container Style Queries into custom properties and standard properties

### ISSUE THREAD

**Issue by woody-li** [Fyrd/caniuse#6842]
The first standard of Container Style Queries is CSS Containment 3.
But there's some questions for standard properties usage: https://github.com/w3c/csswg-drafts/issues/7185

Chrome implementd it only for custom properties.
Webkit expressed support position fot custom properties: https://github.com/WebKit/standards-positions/issues/57
MDN data only shows the custom properties usage.

So is it necessary to split it into the two pages?

**Comment by jensimmons** [comment]
I'm not sure that Style Queries for standard properties will ever happen. It's likely it will not.

**Comment by SebastianZ** [comment]
> I'm not sure that Style Queries for standard properties will ever happen. It's likely it will not.

The [resolution from last week](https://github.com/w3c/csswg-drafts/issues/7185#issuecomment-1869931358) indicates that they'll be added to level 4.

Anyway, in regard of the resolution, I suggest to just change the feature title and description to note that they only apply to custom properties for now.
I.e. change the title to "CSS Container Style Queries for custom properties" and the description to "Style queries in Container Queries provide a way to query the current styling defined in custom properties of a container, and conditionally apply additional CSS to the contents of that container."

Once Style Queries for standard properties become a thing, a new entry can be added to cover them.

Sebastian

**Comment by woody-li** [comment]
What's more, Safari 18 will support it: https://developer.apple.com/documentation/safari-release-notes/safari-18-release-notes#CSS

**Comment by jensimmons** [comment]
Yes, Safari 18 beta supports CSS Container Style Queries for custom properties. We are supportive of the idea of making that page its own feature, since support for all properties is separate in the specification.

**Comment by mirisuzanne** [comment]
For the foreseeable future, style queries will be only for custom properties _in all browsers_. I do think non-custom property queries should be treated as a distinct feature, currently with no support. And the browsers that support custom property queries should show full support for that feature. Since the browsers are treating these as distinct features, authors are as well, and support data should reflect that.

**Comment by woody-li** [comment]
Firefox 151 supports both custom properties and standard properties.
So, should we close it as not planned?

**Comment by SebastianZ** [comment]
> Firefox 151 supports both custom properties and standard properties.

No, it doesn't. It currently only supports custom properties like any other browser.

Simple example:
https://codepen.io/editor/SebastianZ1983/pen/019f4e31-6c32-7217-8025-dc5ea8139d41

Sebastian

**Comment by woody-li** [comment]
@SebastianZ Thanks for your additional info.

But the [release notes](https://developer.mozilla.org/en-US/docs/Mozilla/Firefox/Releases/151#css) said that `This allows you to check if a container has a valid CSS declaration, a CSS property, or a custom property, and apply styles to its children accordingly`.

And in #7529 , it's marked as fully supported.

**Comment by SebastianZ** [comment]
> But the [release notes](https://developer.mozilla.org/en-US/docs/Mozilla/Firefox/Releases/151#css) said that `This allows you to check if a container has a valid CSS declaration, a CSS property, or a custom property, and apply styles to its children accordingly`.

Yeah, this note is incorrect. I've just filed a bug report for that. See https://github.com/mdn/content/issues/44701.

> And in https://github.com/Fyrd/caniuse/pull/7529 , it's marked as fully supported.

I assume that's derived from those release notes.

See also https://developer.mozilla.org/en-US/docs/Web/CSS/Guides/Containment/Container_size_and_style_queries#container_style_queries:~:text=The%20only%20style%20feature%20currently%20supported%20is%20custom%20properties%2C%20with%20or%20without%20a%20value%2E, https://developer.chrome.com/docs/css-ui/style-queries#:~:text=However%2C%20in%20the%20rollout%20of%20this%20feature%2C%20style%20queries%20currently%20only%20work%20with%20CSS%20custom%20property%20values%2E, and https://webkit.org/blog/15865/webkit-features-in-safari-18-0/#:~:text=WebKit%20for%20Safari%2018%2E0%20adds%20support%20for%20Style%20Queries%20when%20testing%20CSS%20Custom%20Properties%2E.

Sebastian
